frontend/src/components/navbar/UserSettingsButton.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useContext } from "react";
+import { IconButton } from "@mui/material";
+import AccountCircle from "@mui/icons-material/AccountCircle";
+
+import { ShoppingCartContext } from "../../contexts/shopping-cart-context/shoppingCartContext";
+
+interface UserSettingMenuProps {
+    userName?: string;
```

Comment:
it shouldn't be optional, right?
`  userName: string;`

Reply:
That was done by mistake, thanx.